git/repo/base.py

```diff
@@ -50,8 +51,11 @@
 __all__ = ('Repo',)
 
 
-def _expand_path(p):
-    return osp.normpath(osp.abspath(osp.expandvars(osp.expanduser(p))))
+def _expand_path(p, unsafe=True):
```

Comment:
Better call it what it is, `expand_vars`, and explain the reason for using this flag in the invoking site.

And I would split it, not to have to compare lines to discover the difference:
```python
p = osp.expanduser(p)
if expand_vars:
    p = osp.expandvars(p)
return osp.normpath(osp.abspath(p))
```